### PR TITLE
feat(esl-mixin-element): v2.0.0 improvements

### PR DESCRIPTION
**Pull request by ala-n**
- Closes: #1560 
- Updates and fixes of the `ESLMixinELement` API

### CODE REVIEW

**Review by fshovchko — APPROVED** [review]

File: src/modules/esl-mixin-element/ui/esl-mixin-attr.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  private static instanceFor(mixin: ESLMixinElement): ESLMixinAttributesObserver | null {
+    const {is, observedAttributes} = mixin.constructor as typeof ESLMixinElement;
+    const attributes = (observedAttributes || []).filter((name: string) => name !== is);
+    if (!is || !attributes.length) return null;
+    return new ESLMixinAttributesObserver(is);
+  }
+
+  /** Subscribe to the {@link ESLMixinElement} host instance mutations */
```
Comment (NastaLeo) [inline src/modules/esl-mixin-element/ui/esl-mixin-attr.ts:58]:
```suggestion
  /** Subscribes to the {@link ESLMixinElement} host instance mutations */
```

File: src/modules/esl-mixin-element/ui/esl-mixin-attr.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  protected observer = new MutationObserver(
+    (records: MutationRecord[]) => records.forEach(this.handleRecord, this)
+  );
+
+  private constructor(protected readonly type: string) {
+    if (instances.has(type)) return instances.get(type)!;
+    instances.set(type, this);
+  }
+
+  /** Process single mutation record */
```
Comment (NastaLeo) [inline src/modules/esl-mixin-element/ui/esl-mixin-attr.ts:22]:
```suggestion
  /** Processes single mutation record */
```

File: src/modules/esl-mixin-element/ui/esl-mixin-attr.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  protected handleRecord(record: MutationRecord): void {
+    const name = record.attributeName;
+    const target = record.target as HTMLElement;
+    if (!name || !target) return;
+    const mixin = ESLMixinRegistry.get(target, this.type) as ESLMixinElementInternal;
+    mixin && mixin.attributeChangedCallback(name, record.oldValue, target.getAttribute(name));
+  }
+
+  /** Subscribe to the {@link ESLMixinElement} host instance mutations */
```
Comment (NastaLeo) [inline src/modules/esl-mixin-element/ui/esl-mixin-attr.ts:31]:
```suggestion
  /** Subscribes to the {@link ESLMixinElement} host instance mutations */
```

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-mixin-element/ui/esl-mixin-attr.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import {ESLMixinRegistry} from './esl-mixin-registry';
+import type {ESLMixinElement, ESLMixinElementInternal} from './esl-mixin-element';
+
+// Singleton cache for ESLMixinAttributesObserver instances
+const instances = new Map<string, ESLMixinAttributesObserver>();
+
+/**
+ * Internal {@link ESLMixinElement}s observedAttributes mutation listener.
+ * Creates a single instance per mixin type
+ * Ignores mixin primary attribute changes (they are observed by {@link ESLMixinRegistry} ootb)
```
Comment (yadamskaya) [inline src/modules/esl-mixin-element/ui/esl-mixin-attr.ts:10]: don't understand what _ootb_ means, please explain me

**Review by yadamskaya — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

**Review by Natalie-Smirnova — APPROVED** [review]

File: src/modules/esl-mixin-element/ui/esl-mixin-attr.ts (hunk 4 above)
Comment (ala-n) [reply src/modules/esl-mixin-element/ui/esl-mixin-attr.ts:10]: out of the box, but not sure is that correct to write it lowercase

**Review by abarmina — APPROVED** [review]

**Pull request merged**